### PR TITLE
build(deps): replace tendermint/tm-db with cometbft/cometbft-db

### PR DESCRIPTION
**Pull request by lklimek**
## Issue being fixed or feature implemented

Build of e2e tests fail due to issues with `github.com/dgraph-io/badger/v2/table`, which is used by  `tendermint/tm-db`:

```
go mod why github.com/dgraph-io/badger/v2/table 
# github.com/dgraph-io/badger/v2/table
github.com/dashpay/tenderdash/abci/example/kvstore
github.com/tendermint/tm-db
github.com/dgraph-io/badger/v2
github.com/dgraph-io/badger/v2/table
```

See https://github.com/dashpay/tenderdash/actions/runs/11662276047/job/32468304503?pr=972: 

```
 #20 18.67 # github.com/dgraph-io/badger/v2/table
Error: #20 18.67 /go/pkg/mod/github.com/dgraph-io/badger/v2@v2.2007.4/table/table.go:79:14: cannot use generic type ristretto.Cache[K z.Key, V any] without instantiation
Error: #20 18.67 /go/pkg/mod/github.com/dgraph-io/badger/v2@v2.2007.4/table/table.go:80:14: cannot use generic type ristretto.Cache[K z.Key, V any] without instantiation
```

Tendermint seems abandoned, so we switch to CometBFT fork of tm-db.

## What was done?

replace github.com/tendermint/tm-db with github.com/cometbft/cometbft-db

Updated golang to 1.23.2 as cometbft-db requires at least 1.23.1

## How Has This Been Tested?

GHA passing

## Breaking Changes

None - it only affects tests:

```
go mod why github.com/cometbft/cometbft-db 
# github.com/cometbft/cometbft-db
github.com/dashpay/tenderdash/abci/example/kvstore
github.com/cometbft/cometbft-db
```

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

## Summary by CodeRabbit

- **New Features**
	- Updated database management library to enhance performance and compatibility.
  
- **Bug Fixes**
	- Improved error handling during database operations across various modules.

- **Documentation**
	- Revised style guide to clarify library imports, dependency management, and testing standards.
	- Enhanced tutorials for building applications in Go, specifying updated Go version requirements.

- **Tests**
	- Adjusted test cases to utilize the new database library while preserving existing logic and functionality.

- **Chores**
	- Updated import paths throughout the codebase to reflect the new database library.
	- Updated Go version in workflow configurations and Dockerfiles to 1.23.2.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module github.com/dashpay/tenderdash
 
-go 1.23
+go 1.23.1
+
+toolchain go1.23.2
```
[the remark this answers is not in the document]
Comment (lklimek) [reply go.mod]: Looks like dependencies need go 1.23.1; I will update the go version in CI/CD to be at minimum 1.23.1.

File: internal/state/indexer/block/kv/kv_test.go [hunk 2]
```diff
@@ -5,8 +5,8 @@ import (
 	"fmt"
 	"testing"
 
+	dbm "github.com/cometbft/cometbft-db"
```
[the remark this answers is not in the document]
Comment (lklimek) [reply internal/state/indexer/block/kv/kv_test.go:8]: We use  v1.0.1

File: light/store/db/db_test.go [hunk 3]
```diff
@@ -7,9 +7,9 @@ import (
 
 	sync "github.com/sasha-s/go-deadlock"
 
+	dbm "github.com/cometbft/cometbft-db"
```
[the remark this answers is not in the document]
Comment (lklimek) [reply light/store/db/db_test.go:10]: we use 1.0.1, updated

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -19,7 +19,7 @@ jobs:
     steps:
       - uses: actions/setup-go@v5.1.0
         with:
-          go-version: "1.23"
+          go-version: "1.23.2"
```
[the remark this answers is not in the document]
Comment (lklimek) [reply .github/workflows/tests.yml:22]: go 1.23.2 was already released.

**Review by shumkov — APPROVED** [review]
👍

**Pull request merged**